src/python/grpcio_tests/tests_aio/unit/metadata_test.py

```diff
@@ -291,6 +310,12 @@ async def test_compatibility_with_tuple(self):
         self.assertEqual(expected_sum, metadata_obj + aio.Metadata(
             ('third', '3')))
 
+    async def test_inspect_context(self):
+        multicallable = self._client.unary_unary(_TEST_INSPECT_CONTEXT)
+        call = multicallable(_REQUEST)
+        with self.assertRaises(grpc.RpcError):
+            await call
```

Comment:
The goal of this test is to ensure that the status code can be read from the server-side context object; we really don't care what the error is on the client side for the purpose of this test. IMO that would make the test less readable, since it would be unclear if what matters in the context of the test is the server-side status code, or the client-side status code.

Reply:
Without checking the error on the main test thread (where the client is), no errors on the server side will propagate to the main thread in such a way that the test will fail. Try deleting some of the `set_XXX` calls in the server handler -- the test won't currently fail.